docfx_yaml/extension.py

```diff
@@ -1348,6 +1363,61 @@ def pretty_package_name(package_group):
     return " ".join(capitalized_name)
 
 
+# Type alias used for yaml entries.
+_yaml_type_alias = dict[str, any]
+
+
+def _find_summary_details(
+    yaml_data: _yaml_type_alias,
+    summary_type: str,
+    cgc_url: str,
+) -> _yaml_type_alias:
+    """Finds the summary details to add for a given entry."""
+    uid = yaml_data.get("uid", "")
+    item_to_add = uid if summary_type == CLASS else f"{uid}-summary"
+
+    if item_to_add not in _ENTRY_NAME_AND_ENTRY_CONTENT_BY_SUMMARY_TYPE[summary_type][0]:
```

Comment:
I think you can avoid this check if your use a `Set` for your `uid`s.

Reply:
I think the problem was that I ran into the same items twice, not that the UIDs are not unique :/ I was seeing duplicate entries for each entry, but the UIDs are definitely unique (google.cloud.package.module.item)